### PR TITLE
16066: Refactor Plan Shopping code to consider new FA model changes

### PR DESCRIPTION
**Pull request by saikumar9**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [16066](https://devops.dchbx.org/redmine/issues/16066) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-3 |
| Development Story Points | 2 |
| Peer Review Story Points | n/a |
| Ticket Author | @psk86 |

### CODE REVIEW

File: app/controllers/insured/plan_shoppings_controller.rb [hunk 1]
```diff
@@ -163,11 +163,12 @@ def show
     hbx_enrollment_id = params.require(:id)
     @change_plan = params[:change_plan].present? ? params[:change_plan] : ''
     @enrollment_kind = params[:enrollment_kind].present? ? params[:enrollment_kind] : ''
+    @family_member_ids = params[:family_member_ids]
     set_plans_by(hbx_enrollment_id: hbx_enrollment_id)
-    shopping_tax_household = get_shopping_tax_household_from_person(@person, @hbx_enrollment.effective_on.year)
+    shopping_tax_household = get_tax_household_from_family_members(@person, @family_member_ids)
```
Comment (psk86) [inline app/controllers/insured/plan_shoppings_controller.rb]: rename shopping_tax_household to shopping_tax_households as it returns a collection of THHs

File: app/controllers/insured/plan_shoppings_controller.rb [hunk 2]
```diff
@@ -163,11 +163,12 @@ def show
...
     if shopping_tax_household.present? && @hbx_enrollment.coverage_kind == "health" && @hbx_enrollment.kind == 'individual'
       @tax_household = shopping_tax_household
-      @max_aptc = @tax_household.total_aptc_available_amount_for_enrollment(@hbx_enrollment)
+      @max_aptc = total_aptc_on_tax_households(shopping_tax_household, @hbx_enrollment)
```
Comment (psk86) [inline app/controllers/insured/plan_shoppings_controller.rb]: rename shopping_tax_household to shopping_tax_households

File: app/models/plan.rb [hunk 3]
```diff
@@ -451,12 +451,18 @@ def reference_plan_metal_level_for_options
...
+    def individual_plans(coverage_kind:, active_year:, tax_households:)
       case coverage_kind
       when 'dental'
         Plan.individual_dental_by_active_year(active_year).with_premium_tables
       when 'health'
-        csr_kind = tax_household.try(:latest_eligibility_determination).try(:csr_eligibility_kind)
+        csr_kinds = []
+        #TODO select the right csr_kind from the array of csr_kinds.
+        tax_households.each do |tax_household|
+          csr_kinds << tax_household.try(:preferred_eligibility_determination).try(:csr_eligibility_kind)
```
Comment (psk86) [inline app/models/plan.rb]: see if you can remove try

**Comment by lokelokesh** [comment]
This PR changes merged to "data-model-14286"

**Pull request closed without merging**